include/swaybar/config.h

```diff
@@ -65,6 +76,16 @@ uint32_t parse_position(const char *position);
 char *parse_font(const char *font);
 
 /**
+ * Parse display mode dock|hide|invisible.
+ */
+enum display_mode_types parse_display_mode(const char *display_mode);
+
+/**
+ * Parse hidden state show|hide.
+ */
+enum hidden_states parse_hidden_state(const char *hidden_state);
+
```

Comment:
> Macros - fuck no

BS. boilerplate - fuck no.

Nothing eliminates bugs faster than deleting code. Don't be a zealot -- when exceptions make sense, make them.

> sizeof(mode_enums) / sizeof(char *) - no, doesn't work if the array uses anonymous structs 
> (so I'd rather not establish a pattern that doesn't work everywhere)

I don't understand how a `char* enum_strings[]` can "use anonymous structs". We're not talking about generic data structure here. And remember that the function we call type-checks that this is the case.